pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/ArrayBasedDocIdIterator.java

```diff
@@ -36,37 +35,15 @@ public ArrayBasedDocIdIterator(int[] docIds, int searchableLength) {
 
   @Override
   public int next() {
-    if (_currentDocId == Constants.EOF) {
-      return Constants.EOF;
-    }
-    if (++_currentIndex == _searchableLength) {
-      _currentDocId = Constants.EOF;
+    if (_nextIndex < _searchableLength) {
+      return _docIds[_nextIndex++];
```

Comment:
what happens if _nextIndex >= docIds.length? should we have a array length check?

Reply:
Caller should ensure `_searchableLength <= _docIds.length`. If caller passes in wrong parameters, it is okay to through IndexOutOfBoundException. Similarly, we don't perform null check everywhere for simplicity and performance reason.